csharp/src/Ice/Connection.cs

```diff
@@ -677,7 +677,7 @@ public void AsyncRequestCanceled(OutgoingAsyncBase outAsync, System.Exception ex
                         _asyncRequests.Remove(o.RequestId);
                     }
 
-                    if (ex is ConnectionTimeoutException)
+                    if (ex is ConnectionIdleException)
```

Comment:
This isn't correct. The ConnectionTimeoutException check here was to support `invocationTimeout = -2` which is supposed to close the connection on invocation timeout. You changed the invocation timeout cancellation to now use TimeoutException for both `invocationTimeout = -2` and `!= -2` so there's no way to distinguish anymore between these 2 cases. We should either remove support for invocationTimeout = -2 and this `if()` block or allow the code to distinguish between the 2 reasons for invocation timeouts.

Reply:
Fixed by removing `invocationTimeout = -2`. It's no longer a valid value.